dkg/nodesigs.go

```diff
@@ -50,32 +62,64 @@ func newNodeSigBcast(operatorAmt int, bcastComp *bcast.Component) nodeSigBcast {
 }
 
 // broadcastCallback is the default bcast.Callback for nodeSigBcast.
-func (n *nodeSigBcast) broadcastCallback(_ context.Context, _ peer.ID, _ string, msg proto.Message) error {
+func (n *nodeSigBcast) broadcastCallback(ctx context.Context, peer peer.ID, _ string, msg proto.Message) error {
 	response, ok := msg.(*dkgpb.MsgNodeSig)
 	if !ok {
 		return errors.New("invalid node sig type")
 	}
 
-	n.otherSigs <- response
+	sig := response.GetSignature()
+	msgPeerIdx := int(response.GetPeerIndex())
+
+	if (msgPeerIdx == n.nodeIdx.PeerIdx) || (msgPeerIdx < 0 || msgPeerIdx > n.operatorAmt) {
+		return errors.New("wrong peer index", z.Str("peer", peer.String()))
+	}
+
+	peerPubk, err := n.peers[msgPeerIdx].PublicKey()
+	if err != nil {
+		return errors.Wrap(err, "can't get peer public key", z.Str("peer", peer.String()))
+	}
+
+	// wait until lock hash becomes available
+	for {
```

Comment:
this will spin, using 100% CPU. As suggested, rather use a channel and storing the resulting lockhash in `nodeSigBcast` behind a mutex:

```
func (b nodeSigBcast) getLockHash(ctx context.Context) lockHash, error {
  b.mu.Lock()
  defer b.mu.Lock()
  
  if b.lockHash {
    return b.lockHash
  } 
  
  select {
     case <-ctx.Done: return ctx.Err()
     case b.LockHash <- b.lockHashCh: return b.lockHash
  }
}
```